### ISSUE TITLE
Millicast support

### ISSUE THREAD

**Issue by robashton** [webrtc-rs/webrtc#184]
This is more of a placeholder because I'll be sending a pull request to [webrtc-dtls](https://github.com/webrtc-rs/dtls/) shortly.

On connecting to the WHIP endpoint at millicast, I get the charming error

```ring::unspecified```

During flight5 of DTLS negotation. 

It turns out that Millicast doesn't support ECDSA (yet), so we pick one of the RSA algorithms and we actually end up picking the wrong one (we should be picking RSA_PKCS1_2048_8192_SHA256, but we end up with RSA_PKCS1_1024_8192_SHA1_FOR_LEGACY_USE_ONLY).

This *might* well be related to https://github.com/webrtc-rs/dtls/issues/15 although I've not read into that in depth.

Either way, I'll amend the selection code once I've read up on the various OIDs and what the various combinations of them actually mean and send a pull request.

On top of that, I'm wondering if perhaps I should improve some of the error handling while I'm here, bailing out with a generic exception from Ring when I could promote that error to something like 'ErrorDuringKeyVerification' might be nice, I had to add quite a bit of trace before I found out exactly where things were going wrong and why :)

**Comment by robashton** [comment]
Sigh, update: It's not the selection code, we're doing the right thing with the OIDs we've got, somebody else is doing a naughty (hopefully not millicast)

**Comment by robashton** [comment]
Ooft, okay - I'm starting to see it now.

We've got the code that is using

```
    let pki_alg = &certificate.tbs_certificate.subject_pki.algorithm.algorithm;
    let sign_alg = &certificate.tbs_certificate.signature.algorithm;
```

And we're using them to determine the algorithm required for verification - but this is wrong *I think*, see also: https://crypto.stackexchange.com/questions/31479/certificate-signature-algorithm-differs-from-signature-algorithm-negotiated-for

In this case, we've negotiated TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, and been given an appropriate certificate whose *key signature* happens to be SHA-1.

I suspect the appropriate action to take is to use the cipher we negotiated to infer how we do key verification, rather than attempting to inspect the certificate when we already have the answer.
